src/new_index/db.rs

```diff
@@ -26,7 +26,7 @@ impl<'a> Iterator for ScanIterator<'a> {
         if self.done {
             return None;
         }
-        let (key, value) = self.iter.next()?;
+        let (key, value) = self.iter.next()?.expect("valid iterator");
```

Comment:
rust-rocksdb [was changed](https://github.com/rust-rocksdb/rust-rocksdb/pull/648) to iterate over a `Result` that may contain an error if the iteration fails.

Previously the iterator was terminated with a `None` and it was expected that users of the library manually check `iter.status()` for errors after calling `next()`, which we did not do -- meaning that errors were ignored.

https://github.com/mempool/electrs/pull/16 updated rust-rocksdb while keeping the old behavior, but it seems more correct to panic on errors as I implemented here.